src/internal/calculateX96PriceFromAmounts.ts

```diff
@@ -0,0 +1,7 @@
+export function calculateX96PriceFromAmounts(amount0: bigint, amount1: bigint): bigint {
+  if(amount1 === 0n) {
+    throw new Error("Division by zero");
+  }
+
+  return (amount0 * (2n ** 96n)) / amount1;
```

Comment:
@mikec could you make sure you want this division. i.e. tokenOut / tokenIn and not the opposite